internal/apis/certmanager/validation/certificate.go

```diff
@@ -331,3 +335,18 @@ func validateAdditionalOutputFormats(crt *internalcmapi.CertificateSpec, fldPath
 
 	return el
 }
+
+func validateKeystores(crt *internalcmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	var el field.ErrorList
+	if crt.Keystores.JKS != nil {
+		if crt.Keystores.JKS.Password != nil && crt.Keystores.JKS.PasswordSecretRef != nil {
+			el = append(el, field.Invalid(fldPath.Child("keystores", "jks", "password"), crt.Keystores.JKS.Password, "When providing a `PasswordSecretRef` no `Password` properties may be provided."))
+		}
+	}
+	if crt.Keystores.PKCS12 != nil {
+		if crt.Keystores.PKCS12.Password != nil && crt.Keystores.PKCS12.PasswordSecretRef != nil {
+			el = append(el, field.Invalid(fldPath.Child("keystores", "pkcs12", "password"), crt.Keystores.PKCS12.Password, "When providing a `PasswordSecretRef` no `Password` properties may be provided."))
+		}
+	}
```

Comment:
@rquinio1A
Could you add a test here to make sure that `Password` OR `PasswordSecretRef` is set?
As @SgtCoDFish mentioned, we discussed in our standup that we would like to start with a solution without default passwords.